corporate/lib/stripe.py

```diff
@@ -2401,8 +2573,39 @@ def on_paid_plan(self) -> bool:
 
     @override
     def add_sponsorship_info_to_context(self, context: Dict[str, Any]) -> None:
-        # TBD
-        pass
+        context.update(
+            realm_org_type=self.remote_server.org_type,
+            sorted_org_types=sorted(
+                (
+                    [org_type_name, org_type]
+                    for (org_type_name, org_type) in Realm.ORG_TYPES.items()
+                    if not org_type.get("hidden")
+                ),
+                key=self.sponsorship_org_type_key_helper,
+            ),
+        )
+
+    @override
+    def get_sponsorship_request_session_specific_context(
+        self,
+    ) -> SponsorshipRequestSessionSpecificContext:
+        return SponsorshipRequestSessionSpecificContext(
+            realm_user=None,
+            user_info=SponsorshipApplicantInfo(
+                name=self.remote_user_data.full_name,
+                email=self.remote_user_data.email,
+                # TODO: Set user_role when determining which set of users can access the page.
```

Comment:
Anyone who can authenticate as the remote server can access the page; I'll tweak this comment.